src/Lykke.AlgoStore.Api/Controllers/AlgoInstanceHistoryController.cs

```diff
@@ -52,7 +55,18 @@ public async Task<IActionResult> GetHistoryFunctions([FromQuery][Required]string
                     return BadRequest(ErrorResponse.Create(ModelState));
                 }
 
-                var functions = await _service.GetFunctionsAsync(instanceId, fromMoment.ToUniversalTime(), toMoment.ToUniversalTime(), ModelState);
+                var clientId = User.GetClientId();
+
+                var data = await _algoInstancesService.GetAlgoInstanceDataAsync(clientId, instanceId);
```

Comment:
Shouldn't this validation logic be part of the service instead of the controller?

Reply:
Moved.